samples/user-interface/appwidgets/src/main/java/com/example/platform/ui/appwidgets/glance/weather/WeatherGlanceWidget.kt

```diff
@@ -98,10 +109,10 @@ class WeatherGlanceWidget : GlanceAppWidget() {
                 is WeatherInfo.Available -> {
                     // Based on the size render different UI
                     when (size) {
-                        thinMode -> WeatherThin(weatherInfo)
-                        smallMode -> WeatherSmall(weatherInfo)
-                        mediumMode -> WeatherMedium(weatherInfo)
-                        largeMode -> WeatherLarge(weatherInfo)
+                        thinMode -> WeatherThin(weatherInfo.value as WeatherInfo.Available)
```

Comment:
this is probably because it cannot do the safe cast on the when since weatherInfo.value might change in between. Normally you could create a temporary variable
```
when (val currentWeather = weatherInfo.value) {
...
}
```